### PR TITLE
Create enum for codepages

### PR DESCRIPTION
**Pull request by Gymnasiast**
Should achieve the same as #16970, but introducing a nice enum, namespace and proper naming conventions.

### CODE REVIEW

**Review by janisozaur — APPROVED** [review]
Looks good, but Tulio should review as well, as he originally requested those changes

File: src/openrct2/core/String.hpp [hunk 1]
```diff
@@ -26,18 +26,18 @@ using u8string_view = std::basic_string_view<utf8>;
...
-#undef CP_UTF8
-
-    constexpr int32_t CP_932 = 932;    // ANSI/OEM Japanese; Japanese (Shift-JIS)
-    constexpr int32_t CP_936 = 936;    // ANSI/OEM Simplified Chinese (PRC, Singapore); Chinese Simplified (GB2312)
-    constexpr int32_t CP_949 = 949;    // ANSI/OEM Korean (Unified Hangul Code)
-    constexpr int32_t CP_950 = 950;    // ANSI/OEM Traditional Chinese (Taiwan; Hong Kong SAR, PRC); Chinese Traditional (Big5)
-    constexpr int32_t CP_1252 = 1252;  // ANSI Latin 1; Western European (Windows)
-    constexpr int32_t CP_UTF8 = 65001; // Unicode (UTF-8)
-} // namespace CODE_PAGE
+    enum CodePage : int32_t
```
Comment (tupaschoal) [inline src/openrct2/core/String.hpp:31]: Any reason for not doing it an `enum class`?

**Review by tupaschoal — COMMENTED** [review]
This looks good to me, I have just one comment.

File: src/openrct2/core/String.hpp (hunk 1 above)
Comment (Gymnasiast) [reply src/openrct2/core/String.hpp:31]: Yes, it is used as a parameter to Windows API functions quite often.

**Pull request merged**